homeassistant/components/overkiz/sensor.py

```diff
@@ -472,6 +537,20 @@ def native_value(self) -> StateType:
 
         return state.value
 
+    @property
+    def native_unit_of_measurement(self) -> str | None:
+        """Return the unit of measurement."""
+        state = self.device.states.get(self.entity_description.key)
+
+        if not state or not state.value:
+            return None
+
+        if self.entity_description.native_unit_of_measurement_fn:
+            return self.entity_description.native_unit_of_measurement_fn(
+                self.device, state.name
+            )
+        return None
```

Comment:
I think that you don't need to change any of the `SensorDescription`.
It would be much easier to say:
```suggestion
        if not (default_unit := self.entity_description.native_unit_of_measurement):
            return None

        state = self.device.states.get(self.entity_description.key)
        if not state or not state.value:
            return None

        attrs = self.device.attributes
        if (unit := attrs[f"{state.name}MeasuredValueType"]) and (
            unit_value := unit.value_as_str
        ):
            return OVERKIZ_UNIT_TO_HA.get(unit_value, default_unit)

        if (unit := attrs[OverkizAttribute.CORE_MEASURED_VALUE_TYPE]) and (
            unit_value := unit.value_as_str
        ):
            return OVERKIZ_UNIT_TO_HA.get(unit_value, default_unit)

        return default_unit
```

Reply:
Thx for this suggestion. It indeed ease the code. I’ve applied it and test it locally.